beps/00010-event-auditor/README.md

```diff
@@ -0,0 +1,221 @@
+  - Refer to [NIST: Audit and Accountability](https://csrc.nist.gov/projects/cprt/catalog#/cprt/framework/version/SP_800_53_5_1_1/home?element=AU)
+  - Refer to [NIST: Non-Repudiation](https://csrc.nist.gov/projects/cprt/catalog#/cprt/framework/version/SP_800_53_5_1_1/home?element=AU-10)
+- Establish a standardized data format for audit events.
+  - Refer to [NIST: Content of Audit Records](https://csrc.nist.gov/projects/cprt/catalog#/cprt/framework/version/SP_800_53_5_1_1/home?element=AU-03)
+- Provide access to the transport layer for customizable output options.
+
+### Non-Goals
+
+- Implementing mechanisms for event storage, analysis, or visualization.
+- Addressing security aspects of event storage and access control beyond initial separation from regular events.
+
+Both of these non-goals can be implemented as separate plugins.
+
+## Proposal
+
+The proposal introduces a crucial change to implement a dedicated audit event stream in Backstage. We recommend creating a new backend service using Winston to establish a distinct channel specifically for audit events. This service would act as a wrapper around Winston, providing methods with strict interfaces to ensure uniformity across audit events throughout the Backstage application. By separating the configuration, we can clearly distinguish between regular application events and critical security events. The event format would include mandatory fields relevant to regulatory compliance and security investigations, such as actor, IP address, timestamp, and event details. This standardized format would streamline the analysis and investigation of security-related events.
+
+Overall, this approach offers several benefits. Separating the configuration allows for the clear distinction of security-critical events, enhancing monitoring and analysis. The standardized data format within the service ensures consistency and facilitates compliance with regulations. Finally, the service methods simplify the process of logging audit events.
+
+## Design Details
+
+### Data Model for Audit Events
+
+To ensure consistency and facilitate regulatory compliance, the proposal suggests creating a shared package that defines a data model for audit events. This model consists of several key components.
+
+#### Actor Details Interface
+
+This interface defines the information related to the actor who triggered the logged event. It includes fields like actor ID, IP address, hostname, and user agent.
+
+```ts
+export type ActorDetails = {
+  actorId: string | null;
+  ip?: string;
+  hostname?: string;
+  userAgent?: string;
+};
+```
+
+#### Audit Request/Response Interfaces
+
+These interfaces define the structure of request and response data that might be included in the audit event. It's important to note that these interfaces exclude sensitive information like tokens from headers or other irrelevant details to avoid security risks.
+
+```ts
+export type AuditRequest = {
+  body: any;
```

Comment:
I think the main use case of having `body` is for those cases where an invalid body is posted (can't parse it by a specific schema). This could potentially be dealt with with something like the `openapi-express-validator` middleware for endpoints, but not all routers have one.

I guess this can also be abused by users by sending extremely large request bodies which could pose issues if we include them into the audit log events.